smartsim/_core/mli/infrastructure/control/worker_manager.py

```diff
@@ -303,18 +312,21 @@ def _on_iteration(self) -> None:
                 self._perf_timer.measure_time("serialize_resp")
 
                 if request.callback:
-                    request.callback.send(serialized_resp)
-                    if reply.has_outputs:
-                        # send tensor data after response
-                        for output in reply.outputs:
-                            request.callback.send(output)
+                    try:
+                        request.callback.send(serialized_resp)
+                        if reply.has_outputs:
+                            # send tensor data after response
+                            for output in reply.outputs:
+                                request.callback.send(output, timeout=None)
```

Comment:
Should we create a new ticket to come back and try to avoid the serial nature of these calls? e.g. Should we be performing the callbacks in a separate thread? The `None` timeout feels like one channel could cause delays everywhere else, but even just thinking that a batch of 100 requires the 100th callback to wait for 99 to complete makes me itch.